Upstream MLIR change introduced a Windows build break

An upstream [MLIR change](https://github.com/llvm/llvm-project/commit/6569cf2a44bf95106e7168bdb79c4674742708fa) introduced a dependency on `llvm::ThreadPool`. The problem is that this thread pool is currently [disabled on Windows to workaround an LLVM bug](https://github.com/tensorflow/runtime/blob/master/third_party/llvm/llvm.bzl#L298-L301).

```
ERROR: external/llvm-project/mlir/BUILD:5066:10: Linking external/llvm-project/mlir/mlir-linalg-ods-yaml-gen.exe [for host] failed: (Exit 1120): link.exe failed: error executing command C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/HostX64/x64/link.exe @bazel-out/host/bin/external/llvm-project/mlir/mlir-linalg-ods-yaml-gen.exe-2.params
IR.lib(Verifier.obj) : error LNK2019: unresolved external symbol "public: bool __cdecl llvm::ThreadPool::isWorkerThread(void)const " (?isWorkerThread@ThreadPool@llvm@@QEBA_NXZ) referenced in function "struct mlir::LogicalResult __cdecl mlir::failableParallelForEach<class mlir::Operation * *,class <lambda_1e2c9a1d5fe91ce52e6f54677d3c54a3> >(class mlir::MLIRContext *,class mlir::Operation * *,class mlir::Operation * *,class <lambda_1e2c9a1d5fe91ce52e6f54677d3c54a3> 
&&)" (??$failableParallelForEach@PEAPEAVOperation@mlir@@V<lambda_1e2c9a1d5fe91ce52e6f54677d3c54a3>@@@mlir@@YA?AULogicalResult@0@PEAVMLIRContext@0@PEAPEAVOperation@0@1$$QEAV<lambda_1e2c9a1d5fe91ce52e6f54677d3c54a3>@@@Z)
bazel-out\host\bin\external\llvm-project\mlir\mlir-linalg-ods-yaml-gen.exe : fatal error LNK1120: 1 unresolved externals
Target //tools:bef_executor failed to build
```

--- a/documents/explicit_dependency.md
+++ b/documents/explicit_dependency.md
@@ -37,7 +37,7 @@ of `Chain`. `Chain` is an empty struct defined in
 as follows:
 
 ```c++
-struct Chain {};
+class Chain {};
 ```
 
 The sole purpose of `Chain` is to add dependencies between kernels that

--- a/include/tfrt/host_context/chain.h
+++ b/include/tfrt/host_context/chain.h
@@ -31,7 +31,7 @@
 
 namespace tfrt {
 
-struct Chain {};
+class Chain {};
 
 class ReadyChain {
  public:

--- a/include/tfrt/host_context/diagnostic.h
+++ b/include/tfrt/host_context/diagnostic.h
@@ -33,7 +33,8 @@ namespace tfrt {
 class ExecutionContext;
 
 // This is a simple representation of a decoded diagnostic.
-struct DecodedDiagnostic {
+class DecodedDiagnostic {
+ public:
   // TODO(b/169618466): carry error code in llvm::Error.
   explicit DecodedDiagnostic(const Error& error);
   explicit DecodedDiagnostic(string_view message) : message(message) {}

--- a/include/tfrt/tensor/tensor_metadata.h
+++ b/include/tfrt/tensor/tensor_metadata.h
@@ -26,7 +26,8 @@ namespace tfrt {
 
 // The metadata of a rectangular tensor that can be computed by a metadata
 // function.
-struct TensorMetadata {
+class TensorMetadata {
+ public:
   TensorMetadata() : shape({}), dtype() {}
 
   TensorMetadata(DType dtype, const TensorShape& shape)

